src/main/java/org/folio/holdingsiq/service/impl/ConfigurationServiceCache.java

```diff
@@ -59,11 +56,12 @@ public CompletableFuture<List<ConfigurationError>> verifyCredentials(Configurati
   }
 
   private CompletableFuture<Void> storeConfigurationInCache(Configuration config, OkapiData okapiData) {
-    return TokenUtils.fetchUserInfo(okapiData.getApiToken())
-      .thenAccept(userInfo -> configurationCache.putValue(
-          userInfo.getUserId(),
-          config.toBuilder().configValid(true).build())
-      );
+    if (okapiData.getUserId() == null || okapiData.getUserId().isEmpty()) {
+      return CompletableFuture.failedFuture(new IllegalArgumentException("User id is empty"));
```

Comment:
Would be nice to add logs on failures, because issue with null userId being put in cache was tricky to discover